File: apps/api/internal/configuration/validate.go
package configuration

// validate enforces the same rules the frontend pre-validates for instant
// feedback — this is the authoritative check.
func validate(req Request) map[string]string {
	errs := map[string]string{}

	if req.NutritionCaloriesTarget <= 0 {
		errs["nutritionCaloriesTarget"] = "must be positive"
	}
	if req.NutritionProteinTarget <= 0 {
		errs["nutritionProteinTarget"] = "must be positive"
	}
	if req.NutritionCarbsTarget <= 0 {
		errs["nutritionCarbsTarget"] = "must be positive"
	}
	if req.NutritionFatTarget <= 0 {
		errs["nutritionFatTarget"] = "must be positive"
	}

	return errs
}

File: apps/web/src/features/configuration/validate.ts
import type { Configuration } from "@/data/schemas/configuration";

// Frontend pre-validates for instant feedback — the Go handler
// (internal/configuration/validate.go) is the authoritative check.
export function validate(input: Configuration): Record<string, string> {
  const errors: Record<string, string> = {};

  if (input.nutritionCaloriesTarget <= 0) errors.nutritionCaloriesTarget = "must be positive";
  if (input.nutritionProteinTarget <= 0) errors.nutritionProteinTarget = "must be positive";
  if (input.nutritionCarbsTarget <= 0) errors.nutritionCarbsTarget = "must be positive";
  if (input.nutritionFatTarget <= 0) errors.nutritionFatTarget = "must be positive";

  return errors;
}
